pkg/resources/mutation/cluster/provisioningcluster.go

```diff
@@ -2,44 +2,89 @@ package cluster
 
 import (
 	"fmt"
+	"reflect"
+	"strings"
 
+	"github.com/blang/semver"
+	v1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
 	"github.com/rancher/webhook/pkg/admission"
+	"github.com/rancher/webhook/pkg/auth"
+	"github.com/rancher/webhook/pkg/clients"
+	v3 "github.com/rancher/webhook/pkg/generated/controllers/management.cattle.io/v3"
 	objectsv1 "github.com/rancher/webhook/pkg/generated/objects/provisioning.cattle.io/v1"
-	"github.com/rancher/webhook/pkg/resources/mutation"
+	"github.com/rancher/webhook/pkg/patch"
+	psa "github.com/rancher/webhook/pkg/podsecurityadmission"
+	"github.com/rancher/wrangler/pkg/data/convert"
+	v1controller "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
 	admissionv1 "k8s.io/api/admission/v1"
 	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/utils/trace"
 )
 
+const (
+	KubeAPIAdmissionConfigOption = "admission-control-config-file"
+	ControlPlaneRoleLabel        = "rke.cattle.io/control-plane-role"
+	SecretAnnotation             = "rke.cattle.io/object-authorized-for-clusters"
+	SecretNamespace              = "fleet-default"
```

Comment:
I don't think the concept of a dedicated secret namespace is applicable here: The secret namespace should be equivalent to the namespace of the associated provisioning cluster. This is currently `"fleet-default"` in practice, but there is active work being done to change this.

Reply:
Good point! 
I updated the code to get the namespace from the cluster object to create the secret in the same namespace as the cluster. Currently, its value is fleet-default.